public/app/plugins/datasource/loki/queryUtils.ts

```diff
@@ -32,55 +33,67 @@ import { LokiDatasource } from './datasource';
 import { getStreamSelectorPositions, NodePosition } from './modifyQuery';
 import { LokiQuery, LokiQueryType } from './types';
 
-export function formatQuery(selector: string | undefined): string {
-  return `${selector || ''}`.trim();
-}
-
 /**
  * Returns search terms from a LogQL query.
  * E.g., `{} |= foo |=bar != baz` returns `['foo', 'bar']`.
  */
-export function getHighlighterExpressionsFromQuery(input: string): string[] {
+export function getHighlighterExpressionsFromQuery(input = ''): string[] {
   const results = [];
 
   const filters = getNodesFromQuery(input, [LineFilter]);
 
-  for (let filter of filters) {
+  for (const filter of filters) {
     const pipeExact = filter.getChild(Filter)?.getChild(PipeExact);
     const pipeMatch = filter.getChild(Filter)?.getChild(PipeMatch);
-    const string = filter.getChild(String);
+    const strings = getStringsFromLineFilter(filter);
```

Comment:
If we want to be very exact, we need to get every string that is not part of an `ip` filter. Because e.g. this would kinda be a mismatch:
![image](https://github.com/grafana/grafana/assets/8092184/8c7e581f-4168-4c18-b1ca-55ae7cac9f73)

`this` is correctly highlighted, while `127.0.0.1` should not be highlighted.

Reply:
Oh gotcha.

Reply:
Should be properly handled now.